elide-core/src/main/java/com/yahoo/elide/security/permissions/expressions/AndExpression.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2015, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.security.permissions.expressions;
+
+import com.yahoo.elide.security.permissions.ExpressionResult;
+
+import static com.yahoo.elide.security.permissions.ExpressionResult.DEFERRED;
+import static com.yahoo.elide.security.permissions.ExpressionResult.FAIL;
+import static com.yahoo.elide.security.permissions.ExpressionResult.PASS;
+
+/**
+ * Representation for an "And" expression.
+ */
+public class AndExpression implements Expression {
+    private final Expression left;
+    private final Expression right;
+
+    /**
+     * Constructor.
+     *
+     * @param left Left expression
+     * @param right Right expression
+     */
+    public AndExpression(final Expression left, final Expression right) {
+        this.left = left;
+        this.right = right;
+    }
+
+    @Override
+    public ExpressionResult evaluate() {
+        ExpressionResult leftResult = left.evaluate();
+        ExpressionResult rightResult = (right == null) ? PASS : right.evaluate();
+
+        if (leftResult == FAIL || rightResult == FAIL) {
+            return FAIL;
+        }
+
+        if (leftResult == PASS && (rightResult == PASS || rightResult == null)) {
```

Comment:
No, this is an artifact of the weird NoopExpression thing I had in there earlier.